vunit.dockerfile

```diff
@@ -1,28 +1,49 @@
 ARG TAG="buster-mcode"
 
+#---
+
 FROM ghdl/ghdl:$TAG AS base
+
 ARG PY_PACKAGES
-RUN apt-get update -qq \
+
+RUN set -ux \
+ && apt-get update -qq \
  && DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
-    ca-certificates \
-    curl \
-    make \
     python3 \
     python3-pip \
  && apt-get autoclean && apt-get clean && apt-get -y autoremove \
- && update-ca-certificates \
- && rm -rf /var/lib/apt/lists/* \
- && pip3 install -U pip setuptools wheel $PY_PACKAGES
+ && rm -rf /var/lib/apt/lists/*
+
+RUN set -ux \
+ && pip3 install --upgrade setuptools wheel $PY_PACKAGES \
+ && rm -rf ~/.cache
+
+#---
 
 FROM base as stable
-RUN pip3 install vunit_hdl
 
-FROM alpine as get-master
-RUN apk add --no-cache --update git && git clone --recurse-submodules https://github.com/VUnit/vunit /tmp/vunit
+RUN set -ux \
+ && pip3 install vunit_hdl \
+ && rm -rf ~/.cache \
+ # Make sure installation worked
+ && python3 -c "import vunit; print(vunit.__version__)"
+
+#---
 
 FROM base AS master
-COPY --from=get-master /tmp/vunit /tmp/vunit
-RUN cd /tmp/vunit \
+
+RUN set -ux \
+ && apt-get update -qq \
```

Comment:
Good catch! I didn't expect VUnit's sources to take so much additional space.

Reply:
Thanks. I discovered while working with a Vivado docker image. The full 35 GB installer was `COPY`'d, and then installed and removed in a `RUN` step. The fact that `COPY` and `RUN` resulted in separate layers was very noticeable in that case. It's a little more subtle in the VUnit case.